### ISSUE TITLE
UpgradeNextMajorParentVersion does not use the new baseline for the Jenkins version

### ISSUE THREAD

**Issue by gounthar** [jenkins-infra/plugin-modernizer-tool#368]
### What feature do you want to see added?

I gave a try to the pipeline-multibranch-defaults plugin with this command:

`java -jar plugin-modernizer-cli/target/jenkins-plugin-modernizer-999999-SNAPSHOT.jar --plugins pipeline-multibranch-defaults --recipes AddOrModernizeJenkinsFile,AddDependencyCheck,MigrateJUnitTestCase,AddPluginsBom,UpgradeNextMajorParentVersion,UpgradeBomVersion,RemoveDependencyVersionOverride,RemoveExtraMavenProperties,UpgradeToJava17,MigrateJUnitTestCase,AddOrModernizeJenkinsFile,AddDependencyCheck`

I got [this output](https://github.com/jenkinsci/pipeline-multibranch-defaults-plugin/compare/master...gounthar:pipeline-multibranch-defaults-plugin:plugin-modernizer-tool#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R36):

 ```
  <properties>
        <jenkins.version>2.479.1</jenkins.version>
    </properties>
[...]
            <dependency>
                <groupId>io.jenkins.tools.bom</groupId>
                <artifactId>bom-2.479.x</artifactId>
                <version>3613.v584fca_12cf5c</version>
                <scope>import</scope>
                <type>pom</type>
            </dependency>
```

We should get something like [this](https://github.com/jenkinsci/git-client-plugin/blob/master/pom.xml#L62):
```
<jenkins.baseline>2.479</jenkins.baseline>
    <jenkins.version>${jenkins.baseline}.1</jenkins.version>
[...]
<dependency>
        <groupId>io.jenkins.tools.bom</groupId>
        <artifactId>bom-${jenkins.baseline}.x</artifactId>
        <version>3613.v584fca_12cf5c</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
```

**Comment by jonesbusy** [comment]
This must be implemented on https://github.com/openrewrite/rewrite-jenkins/blob/main/src/main/java/org/openrewrite/jenkins/AddPluginsBom.java

There is already some logic to detect the `${jenkins.baseline}`. So doing the migration should be easy I guess

1) If bom is not present, add directly with the placeholder
2) If bom is present, perform the migration if not using the placeholder

**Comment by jonesbusy** [comment]
Fixed by #513 

Added on some recipes to migrate to the `jenkins.baseline` 

![Screenshot from 2024-12-26 08-04-53](https://github.com/user-attachments/assets/38084b84-710c-4727-a1ed-008528dbd1b4)
